### PR TITLE
correction filing (part 3)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#2654

*Description of changes:*
This is Part 3 of the the Correction Filing task, which includes:
- implemented loading draft correction
- renamed some route params to correctedFilingId
- blocked correcting a correction (for now)
- enabled Save for corrections
- changed Save buttons to always be enabled (in all filings, per Scott)
- changed some strings to Filing Types
- added some explicit types

Part 4 will implement:
- updates to unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/AnnualReport.vue [hunk 1]
```diff
@@ -208,15 +208,15 @@
       <div class="buttons-left">
         <v-btn id="ar-save-btn" large
           v-if="isAnnualReportEditable"
-          :disabled="!isSaveButtonEnabled || busySaving"
```
Comment (cameron-freshworks) [inline src/views/AnnualReport.vue]: Just curious how this ties in here?

Comment (severinbeauvais) [reply src/views/AnnualReport.vue]: I had to think about this for Correction filing so I asked Scott what the requirement is and whether it applies to all filings. He said you should always be allows to save a draft, for any filing. So I updated it.

Comment (severinbeauvais) [reply src/views/AnnualReport.vue]: I will be sure to add test notes to the ticket so that QA knows to look at it.

**Comment by thorwolpert** [comment]
Will merge with the unit test temporarily failing.

**Pull request merged**